pkg/dao/repository.go

```diff
@@ -84,29 +107,47 @@ func (r repositoryDaoImpl) fetchRepoConfig(orgID string, uuid string) (models.Re
 	return found, nil
 }
 
+func (r repositoryDaoImpl) fetchRepositoryWithURL(url string) (models.Repository, error) {
+	found := models.Repository{}
+	if err := r.db.Where("URL = ?", url).First(&found).Error; err != nil {
+		return found, err
+	}
+	return found, nil
+}
+
 func (r repositoryDaoImpl) Update(orgID string, uuid string, repoParams api.RepositoryRequest) error {
+	var repo models.Repository
+	if repoParams.URL != nil {
+		var err error
+		repo.URL = *repoParams.URL
+		if err = r.db.Where("URL = ?", repoParams.URL).FirstOrCreate(&repo).Error; err != nil {
+			return err
+		}
+	}
 	repoConfig, err := r.fetchRepoConfig(orgID, uuid)
 	if err != nil {
 		return err
 	}
 	ApiFieldsToModel(&repoParams, &repoConfig)
-	result := db.DB.Model(&repoConfig).Updates(repoConfig.MapForUpdate())
-	if result.Error != nil {
-		return DBErrorToApi(result.Error)
+	if repoParams.URL != nil {
```

Comment:
i don't /think/ this check should be necessary because if URL is nil, the repository_configuration should fail to save?

Reply:
oh right you moved url to Repository, so it won't fail!  I think repo.UUID would be nil in that case, so its probably still not necessary (i could be wrong though)

Reply:
thanks! I have realized I was not considering empty string here; I have added that; I have added validation at BeforeCreate to validate that no blank is tried for the URL.

About RepositoryConfiguration.RepositoryUUID, it cannot be null in the database neither into the field, and it is a foreign key to a Repository record; I have realized a mistake I did when mapping the Repository fields for updating as I was not including the URL field :s   now it is updated.